### PR TITLE
Test assisted ZTP with converged flow by default

### PR DESCRIPTION
**Pull request by carbonin**
This removes the converged-specific job from master and marks all ZTP jobs explicitly as non-converged in 2.7 and 2.6.

Resolves https://issues.redhat.com/browse/MGMT-11998

cc @eranco74 @osherdp @adriengentil

### CODE REVIEW

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-release-ocm-2.6-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by eranco74** [comment]
If it marks all ZTP jobs explicitly as non-converged in 2.7.
how it's resolving MGMT-11998 - Update release to set converged flow Enabled when testing ZTP ?

**Comment by eranco74** [comment]
/lgtm
note that this did cause some [issues](https://redhat-internal.slack.com/archives/C014N2VLTQE/p1675069864842159) last time it merged, so consider `pj-rehearse max` just to be sure it's safe.

1. In the disconnected job agents aren't showing up, probably because they can't pull the ironic-agent-image.
2. Other jobs failed due to agents not showing up, currently we don't gather logs from nodes that didn't register as agents so we can't really tell what went wrong, this [task](https://issues.redhat.com/browse/MGMT-11712) should fix it.

**Comment by carbonin** [comment]
/pj-rehearse max

Faster than another revert and retry 😅

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-image-service-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by eranco74** [comment]
Crap, multiple jobs failed due to:
```
[2023-02-06 21:53:29] Waiting for operator local-storage-operator to get installed on namespace openshift-local-storage...
```
:face_with_spiral_eyes:

**Comment by osherdp** [comment]
> Crap, multiple jobs failed due to:
> 
> ```
> [2023-02-06 21:53:29] Waiting for operator local-storage-operator to get installed on namespace openshift-local-storage...
> ```
> 
> face_with_spiral_eyes

from subscription:
```
failed to populate resolver cache from source redhat-marketplace/openshift-marketplace: failed to list bundles: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 172.30.248.218:50051: connect: connection refused\"
```
also there are a couple of errors in the [marketplace operator logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/36093/rehearse-36093-pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-ipv4v6-3masters-ocp-411/1622676952538157056/artifacts/e2e-ai-operator-ztp-ipv4v6-3masters-ocp-411/gather-extra/artifacts/pods/openshift-marketplace_marketplace-operator-599877d56f-9xhb8_marketplace-operator_previous.log). Not sure if they are relevant

**Comment by adriengentil** [comment]
I asked to refresh the signatures, I will update you when it's done.

**Comment by carbonin** [comment]
@adriengentil @osherdp is there any reason to retest here yet?

**Comment by adriengentil** [comment]
I did not see any failures related to the LSO operator during the night, let's retry:
/pj-rehearse max
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-image-service-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
/hold

It looks like the converged flow is still getting enabled in some older jobs somehow.

**Comment by carbonin** [comment]
Failures not due to converged:

|Branch|Job                      | Failure reason                  |
|---------|-----------------|---------------------|
|master|`pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-sno-day2-masters` | timeout|
|master|`pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-compact-day2-workers` | timeout|
|master|`pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-compact-day2-workers` | timeout|
|master| `pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp` | devscripts setup failed|
|master|`pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-compact-day2-masters` | used converged (as it should) but timed out after successfully discovering agents|
|master| `pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-capi` | image service failed to become ready|
|master| `periodic-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-3masters-periodic` | looks like it should have passed but it says the step failed. Not due to converged though because the agents were discovered fine|
|2.7| `pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-compact-day2-workers` | devscripts setup failed|
|2.7| `pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-ipv4v6-3masters` | devscripts setup failed|
|2.7| `pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-ipv4v6-sno` | devscripts setup failed|
|2.7| `pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-multiarch-sno-ocp-411` | devscripts setup failed|
|2.7| `pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-hypershift-zero-nodes` | failed communicating with the bm server|
|2.7| `pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-sno-day2-masters` | timeout|
|2.7| `pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-compact-day2-masters` | looks like it should have passed but didn't. Not running converged|
|2.6| `pull-ci-openshift-assisted-service-release-ocm-2.6-e2e-ai-operator-ztp-capi` | devscripts setup failed|

Ran converged, but shouldn't have:

* `pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.6-e2e-ai-operator-ztp-capi`
```
time="2023-02-08T09:58:55Z" level=info msg="The baremetal operator version is 4.11.0-0.ci.test-2023-02-08-084302-ci-op-m65680w6-latest, the minimal version for the converged flow is 4.11.0-0.alpha" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMOUtils).ConvergedFlowAvailable" file="/assisted-service/internal/controller/controllers/bmo_utils.go:50" pkg=baremetal_operator_utils
time="2023-02-08T09:58:55Z" level=info msg="Converged flow enabled: true" func="github.com/openshift/assisted-service/internal/controller/controllers.(*BMOUtils).ConvergedFlowAvailable" file="/assisted-service/internal/controller/controllers/bmo_utils.go:56" pkg=baremetal_operator_utils
```

Failures possibly due to converged:

* `pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected` - no agents registered

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
@eranco74 what is the success criteria here?
If I go through all these and ensure they are not issues with converged is that good enough? Or do we expect them all to pass eventually?

**Comment by eranco74** [comment]
If none of them failed because the agent didn't register let's push

**Comment by carbonin** [comment]
Okay, I'll take a closer look.

The disconnected one has me worried, we may need to do some additionall mirroring magic there

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.6-e2e-ai-operator-ztp-capi pull-ci-openshift-assisted-service-release-ocm-2.6-e2e-ai-operator-ztp-capi

Retesting these since my last commit should cause them to no longer try to run converged.

**Comment by carbonin** [comment]
Almost all of these were LSO not coming up or devscripts-setup failing.
I'll retest to see if we can get any to pass, but will definitely look into the 2.6 jobs and the disconnected one more tomorrow.

**Comment by carbonin** [comment]
For the disconnected job I'm guessing we need another mirror command [here](https://github.com/openshift/assisted-service/blob/445e8646e95c70b74c9131c57945eff1289f73da/deploy/operator/deploy.sh#L40-L43) to mirror the hub release image, but I'm not sure if there's a straight-forward way of getting that image.

Maybe @osherdp knows a way?

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-release-ocm-2.6-e2e-ai-operator-ztp-capi pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.6-e2e-ai-operator-ztp-capi

Rerunning the 2.6 jobs to ensure they don't use converged now

**Comment by carbonin** [comment]
Last two runs confirmed that converged isn't running on the 2.6 jobs even though they failed.

**Comment by osherdp** [comment]
> For the disconnected job I'm guessing we need another mirror command [here](https://github.com/openshift/assisted-service/blob/445e8646e95c70b74c9131c57945eff1289f73da/deploy/operator/deploy.sh#L40-L43) to mirror the hub release image, but I'm not sure if there's a straight-forward way of getting that image.
> 
> Maybe @osherdp knows a way?

I might be missing some information, but for the hub release image there's already a local mirror made by dev-scripts automation:
https://github.com/openshift-metal3/dev-scripts/blob/8fae70a64b1782998a4dcb04953e9400474836a4/oc_mirror.sh#L117

**Comment by carbonin** [comment]
Hm, okay I'll have to try a live debug then

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
Of course the disconnected test failed for some reason other than not finding agents when I was ready to debug it :man_facepalming: 

```
[2023-02-13 20:38:00] uploading: virthost.ostest.test.metalkube.org:5000/ci-op-6jj6ssv9/release sha256:8c6fafbda8b971cb846b03ac67f03dbf2ebd4e80cdffc071670f0d1139b200df 23.58KiB
[2023-02-13 20:38:00] uploading: virthost.ostest.test.metalkube.org:5000/ci-op-6jj6ssv9/release sha256:9020822586c8a031018b1da370a9b0416fb04f62a61441bb26225a5781a15e27 453.7KiB
[2023-02-13 20:38:01] uploading: virthost.ostest.test.metalkube.org:5000/ci-op-6jj6ssv9/release sha256:da314f1f91e1309b4d16ec3e58ac7649574fad70de6a9eef6ecc00ab5261d331 10.93MiB
[2023-02-13 20:38:01] error: unable to open source layer sha256:14f770810f01aeb587534149468ed06d09f920cdf4a65a1a607f5ace83cb0027 to copy to virthost.ostest.test.metalkube.org:5000/ci-op-6jj6ssv9/release: received unexpected HTTP status: 500 Internal Server Error
[2023-02-13 20:38:01] info: Mirroring completed in 1m38.17s (141.1MB/s)
[2023-02-13 20:38:01] error: one or more errors occurred while uploading images
[2023-02-13 20:38:01] +(:1): removetmp
[2023-02-13 20:38:01] +(utils.sh:668): removetmp(): '[' -n '' ']'
[2023-02-13 20:38:01] +(utils.sh:668): removetmp(): true
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"k8s.io/test-infra/prow/entrypoint/run.go:79","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2023-02-13T20:38:01Z"}
error: failed to execute wrapped command: exit status 1 
INFO[2023-02-13T20:38:04Z] Step e2e-ai-operator-ztp-disconnected-assisted-baremetal-operator-setup failed after 2m50s. 
INFO[2023-02-13T20:38:04Z] Step phase pre failed after 57m50s.          
```

Will try again tomorrow.

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
Missed the last one ... was in meetings. Trying again :man_facepalming: 

/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
Failed in setup again:
```
[2023-02-14 20:20:45] install_lso
[2023-02-14 20:20:46] Created project openshift-local-storage
[2023-02-14 20:20:46] Attempt 1/5 to execute "oc annotate namespace openshift-local-storage openshift.io/node-selector= --overwrite=true"...
[2023-02-14 20:20:46] namespace/openshift-local-storage annotated
[2023-02-14 20:20:46] /usr/bin/which: no opm in (/root/go/bin:/usr/local/go/bin:/root/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin)
[2023-02-14 20:20:46] opm
[2023-02-14 20:20:46] operatorhub.config.openshift.io/cluster patched (no change)
[2023-02-14 20:20:46] time="2023-02-14T20:20:46Z" level=info msg="pruning the index" packages="[local-storage-operator]"
[2023-02-14 20:20:46] time="2023-02-14T20:20:46Z" level=info msg="Pulling previous image registry.redhat.io/redhat/redhat-operator-index:v4.8 to get metadata" packages="[local-storage-operator]"
[2023-02-14 20:20:46] time="2023-02-14T20:20:46Z" level=info msg="running /usr/bin/podman pull registry.redhat.io/redhat/redhat-operator-index:v4.8" packages="[local-storage-operator]"
[2023-02-14 20:20:49] time="2023-02-14T20:20:49Z" level=error msg="Trying to pull registry.redhat.io/redhat/redhat-operator-index:v4.8...\nError: copying system image from manifest list: Source image rejected: A signature was required, but no signature exists\n" packages="[local-storage-operator]"
[2023-02-14 20:20:49] Error: error pulling image: Trying to pull registry.redhat.io/redhat/redhat-operator-index:v4.8...
[2023-02-14 20:20:49] Error: copying system image from manifest list: Source image rejected: A signature was required, but no signature exists
[2023-02-14 20:20:49] . exit status 125
[2023-02-14 20:20:49] Usage:
[2023-02-14 20:20:49]   opm index prune [flags]
[2023-02-14 20:20:49] 
[2023-02-14 20:20:49] Flags:
[2023-02-14 20:20:49]   -i, --binary-image opm        container image for on-image opm command
[2023-02-14 20:20:49]   -c, --container-tool string   tool to interact with container images (save, build, etc.). One of: [docker, podman] (default "podman")
[2023-02-14 20:20:49]   -f, --from-index string       index to prune
[2023-02-14 20:20:49]       --generate                if enabled, just creates the dockerfile and saves it to local disk
[2023-02-14 20:20:49]   -h, --help                    help for prune
[2023-02-14 20:20:49]   -d, --out-dockerfile string   if generating the dockerfile, this flag is used to (optionally) specify a dockerfile name
[2023-02-14 20:20:49]   -p, --packages strings        comma separated list of packages to keep
[2023-02-14 20:20:49]       --permissive              allow registry load errors
[2023-02-14 20:20:49]   -t, --tag string              custom tag for container image being built
[2023-02-14 20:20:49] 
[2023-02-14 20:20:49] Global Flags:
[2023-02-14 20:20:49]       --skip-tls   skip TLS certificate verification for container image registries while pulling bundles or index
[2023-02-14 20:20:49] 
```

:tired_face:

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
Got it:
```
[root@extraworker-0 core]# cat /etc/containers/registries.conf
unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]


[[registry]]
  location = "registry.build03.ci.openshift.org/ci-op-d49wt04n/release"
  insecure = false
  mirror-by-digest-only = true

  [[registry.mirror]]
    location = "virthost.ostest.test.metalkube.org:5000/ci-op-d49wt04n/release"


[[registry]]
  location = "registry.build03.ci.openshift.org/ci-op-d49wt04n/stable-stable-4-10"
  insecure = false
  mirror-by-digest-only = true

  [[registry.mirror]]
    location = "virthost.ostest.test.metalkube.org:5000/ci-op-d49wt04n/release"


[[registry]]
  location = "registry.build03.ci.openshift.org/ci-op-d49wt04n/pipeline"
  insecure = false
  mirror-by-digest-only = true

  [[registry.mirror]]
    location = "virthost.ostest.test.metalkube.org:5000/olm/ci-op-d49wt04n-pipeline"
[root@extraworker-0 core]# journalctl -fu ironic-agent
-- Logs begin at Tue 2023-02-14 22:07:41 UTC. --
Feb 14 22:10:21 extraworker-0.ostest.test.metalkube.org podman[1598]: Trying to pull quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:83de8d0b879bf2a8fc32dad0ccb6ba63a261fa8558250d614fd6c8f0d3f3ed22...
Feb 14 22:12:21 extraworker-0.ostest.test.metalkube.org podman[1598]: Error: Error initializing source docker://quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:83de8d0b879bf2a8fc32dad0ccb6ba63a261fa8558250d614fd6c8f0d3f3ed22: error pinging docker registry quay.io: Get "http://quay.io/v2/": dial tcp [2600:1f18:483:cf00:50f5:32dd:7fb9:ece]:80: i/o timeout
Feb 14 22:12:21 extraworker-0.ostest.test.metalkube.org systemd[1]: ironic-agent.service: Control process exited, code=exited status=125
Feb 14 22:12:21 extraworker-0.ostest.test.metalkube.org systemd[1]: ironic-agent.service: Failed with result 'exit-code'.
Feb 14 22:12:21 extraworker-0.ostest.test.metalkube.org systemd[1]: Failed to start Ironic Agent.
Feb 14 22:12:26 extraworker-0.ostest.test.metalkube.org systemd[1]: ironic-agent.service: Service RestartSec=5s expired, scheduling restart.
Feb 14 22:12:26 extraworker-0.ostest.test.metalkube.org systemd[1]: ironic-agent.service: Scheduled restart job, restart counter is at 1.
Feb 14 22:12:26 extraworker-0.ostest.test.metalkube.org systemd[1]: Stopped Ironic Agent.
Feb 14 22:12:26 extraworker-0.ostest.test.metalkube.org systemd[1]: Starting Ironic Agent...
Feb 14 22:12:26 extraworker-0.ostest.test.metalkube.org podman[1802]: Trying to pull quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:83de8d0b879bf2a8fc32dad0ccb6ba63a261fa8558250d614fd6c8f0d3f3ed22...
```

**Comment by carbonin** [comment]
I guess this is more specifically the problem - from assisted service logs:
```
time="2023-02-14T22:06:20Z" level=info msg="Fetching image from OCP release (oc adm release info --image-for=ironic-agent --insecure=false --icsp-file=/tmp/icsp-file3308547216 virthost.ostest.test.metalkube.org:5000/localimages/local-release-image@sha256:12f753e84cd89b716ce306af30b89995a5e0ae50b8bcb422a2a368d61650110f)" func="github.com/openshift/assisted-service/internal/oc.(*release).getImageFromRelease" file="/assisted-service/internal/oc/release.go:287" go-id=687 preprovisioning_image=ostest-extraworker-0 preprovisioning_image_namespace=assisted-spoke-cluster request_id=47c355e6-ef47-4fe4-bf49-d9cc15bff5a4
time="2023-02-14T22:06:20Z" level=info msg="Caching hub ironic agent image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:83de8d0b879bf2a8fc32dad0ccb6ba63a261fa8558250d614fd6c8f0d3f3ed22 for architectures [x86_64]" func="github.com/openshift/assisted-service/internal/controller/controllers.(*PreprovisioningImageReconciler).getIronicAgentImageByRelease" file="/assisted-service/internal/controller/controllers/preprovisioningimage_controller.go:443" go-id=687 preprovisioning_image=ostest-extraworker-0 preprovisioning_image_namespace=assisted-spoke-cluster request_id=47c355e6-ef47-4fe4-bf49-d9cc15bff5a4
```

@osherdp @eranco74 what do we need to handle this? Just something to redirect quay.io to our disconnected registry I guess?

**Comment by carbonin** [comment]
I'm not really sure what's going on in this job.

From the setup logs it looks like we're mirroring a 4.10 release, but the cluster image set we're actually installing is 4.13 and the hub release is a 4.12 nightly based on the service logs.

How does anything work?

**Comment by osherdp** [comment]
> we're mirroring a 4.10 release
https://github.com/openshift/release/blob/a5aba84cbd46e495ede5988ac27c55a4f073b603/ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml#L429

> but the cluster image set we're actually installing is 4.13
maybe this logic is not correct?
https://github.com/openshift/assisted-service/blob/400e890fbaa096253a63fa14f2dae0d99dcda7ac/deploy/operator/common.sh#L46

> and the hub release is a 4.12 nightly
https://github.com/openshift/release/blob/a5aba84cbd46e495ede5988ac27c55a4f073b603/ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml#L430

> How does anything work?
that's always a good question :smile: 

but really, I think we're just missing some mirroring of the internal payload of the image
in this case we need to concentrate on one of the 2 mirror actions we're performing. one of them is in dev-scripts iirc and the other one here:
https://github.com/openshift/assisted-service/blob/47d9fba732e51ed0cfd9f5a768e37e78ae3e52f3/deploy/operator/mirror_utils.sh#L149

it might be that something doesn't function as well as it previously did (maybe related to the dev-scripts supporting both ``oc mirror`` and ``oc adm mirror``?)

**Comment by carbonin** [comment]
> maybe this logic is not correct?
https://github.com/openshift/assisted-service/blob/400e890fbaa096253a63fa14f2dae0d99dcda7ac/deploy/operator/common.sh#L46

In the current master branch this gives me:
```
08:13:53:~/Source/assisted-service (master)$ SORTED_RELEASE_IMAGES=$(cat data/default_release_images.json | jq -rc 'map(select(.cpu_architecture=="x86_64")) | sort_by(.openshift_version|split(".")|map(tonumber))')
08:13:58:~/Source/assisted-service (master)$ VERSION=$(echo ${SORTED_RELEASE_IMAGES} | jq -rc '[.[].openshift_version][-2]')
08:14:05:~/Source/assisted-service (master)$ echo $VERSION
4.12
08:14:11:~/Source/assisted-service (master)$ echo $SORTED_RELEASE_IMAGES | jq
[
  {
    "openshift_version": "4.8",
    "cpu_architecture": "x86_64",
    "cpu_architectures": [
      "x86_64"
    ],
    "url": "quay.io/openshift-release-dev/ocp-release:4.8.55-x86_64",
    "version": "4.8.55"
  },
  {
    "openshift_version": "4.9",
    "cpu_architecture": "x86_64",
    "cpu_architectures": [
      "x86_64"
    ],
    "url": "quay.io/openshift-release-dev/ocp-release:4.9.53-x86_64",
    "version": "4.9.53"
  },
  {
    "openshift_version": "4.10",
    "cpu_architecture": "x86_64",
    "cpu_architectures": [
      "x86_64"
    ],
    "url": "quay.io/openshift-release-dev/ocp-release:4.10.45-x86_64",
    "version": "4.10.45"
  },
  {
    "openshift_version": "4.11",
    "cpu_architecture": "x86_64",
    "cpu_architectures": [
      "x86_64"
    ],
    "url": "quay.io/openshift-release-dev/ocp-release:4.11.20-x86_64",
    "version": "4.11.20"
  },
  {
    "openshift_version": "4.12",
    "cpu_architecture": "x86_64",
    "cpu_architectures": [
      "x86_64"
    ],
    "url": "quay.io/openshift-release-dev/ocp-release:4.12.0-x86_64",
    "version": "4.12.0",
    "default": true
  },
  {
    "openshift_version": "4.13",
    "cpu_architecture": "x86_64",
    "cpu_architectures": [
      "x86_64"
    ],
    "url": "quay.io/openshift-release-dev/ocp-release:4.13.0-ec.2-x86_64",
    "support_level": "beta",
    "version": "4.13.0-ec.2"
  }
]
```

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp
/pj-rehearse pull-ci-openshift-assisted-service-release-ocm-2.5-e2e-ai-operator-ztp
/pj-rehearse pull-ci-openshift-assisted-service-release-ocm-2.6-e2e-ai-operator-ztp

/pj-rehearse pull-ci-openshift-cluster-api-provider-agent-master-e2e-ai-operator-ztp-capi
/pj-rehearse pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.6-e2e-ai-operator-ztp-capi
/pj-rehearse pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.7-e2e-ai-operator-ztp-capi

/pj-rehearse pull-ci-openshift-assisted-installer-agent-release-ocm-2.7-e2e-ai-operator-ztp
/pj-rehearse pull-ci-openshift-assisted-installer-agent-master-edge-e2e-ai-operator-ztp
/pj-rehearse pull-ci-openshift-assisted-installer-agent-release-ocm-2.6-e2e-ai-operator-ztp

/pj-rehearse pull-ci-openshift-assisted-image-service-release-ocm-2.7-e2e-ai-operator-ztp
/pj-rehearse pull-ci-openshift-assisted-image-service-main-edge-e2e-ai-operator-ztp
/pj-rehearse pull-ci-openshift-assisted-image-service-release-ocm-2.6-e2e-ai-operator-ztp

/pj-rehearse pull-ci-openshift-assisted-installer-release-ocm-2.7-e2e-ai-operator-ztp
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp
/pj-rehearse pull-ci-openshift-assisted-installer-release-ocm-2.6-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
I guess it only takes the first one ... so much for formatting :tired_face:

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-release-ocm-2.5-e2e-ai-operator-ztp pull-ci-openshift-assisted-service-release-ocm-2.6-e2e-ai-operator-ztp pull-ci-openshift-cluster-api-provider-agent-master-e2e-ai-operator-ztp-capi pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.6-e2e-ai-operator-ztp-capi pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.7-e2e-ai-operator-ztp-capi pull-ci-openshift-assisted-installer-agent-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-agent-master-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-agent-release-ocm-2.6-e2e-ai-operator-ztp pull-ci-openshift-assisted-image-service-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-image-service-main-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-image-service-release-ocm-2.6-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-release-ocm-2.6-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-release-ocm-2.5-e2e-ai-operator-ztp pull-ci-openshift-assisted-service-release-ocm-2.6-e2e-ai-operator-ztp pull-ci-openshift-cluster-api-provider-agent-master-e2e-ai-operator-ztp-capi pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.6-e2e-ai-operator-ztp-capi pull-ci-openshift-cluster-api-provider-agent-release-ocm-2.7-e2e-ai-operator-ztp-capi pull-ci-openshift-assisted-installer-agent-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-agent-master-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-agent-release-ocm-2.6-e2e-ai-operator-ztp pull-ci-openshift-assisted-image-service-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-image-service-main-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-image-service-release-ocm-2.6-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-release-ocm-2.7-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-installer-release-ocm-2.6-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-disconnected

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-agent-master-edge-e2e-ai-operator-ztp pull-ci-openshift-assisted-service-release-ocm-2.5-e2e-ai-operator-ztp pull-ci-openshift-assisted-service-release-ocm-2.6-e2e-ai-operator-ztp pull-ci-openshift-cluster-api-provider-agent-master-e2e-ai-operator-ztp-capi pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-cluster-api-provider-agent-master-e2e-ai-operator-ztp-capi

**Comment by carbonin** [comment]
I think I'd call this one good to go now.

What do you think @eranco74 @osherdp ?
Any other tests you want to run?

File: ci-operator/config/openshift/cluster-api-provider-agent/openshift-cluster-api-provider-agent-release-ocm-2.6.yaml [hunk 1]
```diff
@@ -117,6 +117,8 @@ tests:
       INDEX_IMAGE: assisted-service-index
       OPENSHIFT_INSTALL_RELEASE_IMAGE: release:latest
       PROVIDER_IMAGE: pipeline:cluster-api-provider-agent
+    env:
+      ALLOW_CONVERGED_FLOW: "False"
```
Comment (osherdp) [inline ci-operator/config/openshift/cluster-api-provider-agent/openshift-cluster-api-provider-agent-release-ocm-2.6.yaml:121]: should be applied to the periodic jobs as well? ^

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by eranco74** [comment]
Tests looks good.
/lgtm

File: ci-operator/config/openshift/cluster-api-provider-agent/openshift-cluster-api-provider-agent-release-ocm-2.6.yaml (hunk 1 above)
Comment (carbonin) [reply ci-operator/config/openshift/cluster-api-provider-agent/openshift-cluster-api-provider-agent-release-ocm-2.6.yaml:123]: Added

**Review by osherdp — APPROVED** [review]

**Pull request merged**